packages/react-instantsearch-hooks/src/components/InstantSearchSSRProvider.tsx

```diff
@@ -22,6 +22,13 @@ export function InstantSearchSSRProvider({
   children,
   ...props
 }: InstantSearchSSRProviderProps) {
+  // When there is dynamic widgets on the page, a second Provider is used
+  // above the user-land InstantSearchSSRProvider. To avoid the user's provider
+  // overriding the value with an empty object, this provider is skipped
```

Comment:
```suggestion
  // When <DynamicWidgets> is mounted, a second provider is used above the user-land
  // <InstantSearchSSRProvider> in `getServerState()`.
  // To avoid the user's provider overriding the context value with an empty object,
  // we skip this provider.
```